File: src/test/resources/yield/Throws.cs
using System.Collections;

namespace yield {
    public sealed class Throws {
        public IEnumerator test() {
            if (true) {
                yield break;
            }
            yield return test2(123);
            yield break;
        }

        public IEnumerator test2(int x) {
            yield break;
        }
    }
}

File: src/test/resources/yield/Throws.java
package yield;

public final class Throws {

    /**
     * @sharpen.ignore
     * @sharpen.yield
     */
    public static class Marker extends Exception {
    }

    public void test() throws Marker {
        if (true) return;
        test2(123);
        return;
    }

    public void test2(int x) throws Marker {
        return;
    }
}
